apps/rush-lib/src/cli/utilities/pnpm/PnpmLinkManager.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import * as semver from 'semver';
+import encodeRegistry = require('encode-registry');
+
+import { JsonFile } from '@microsoft/node-core-library';
+
+import {
+  BaseLinkManager,
+  SymlinkKind
+} from '../base/BaseLinkManager';
+import Utilities from '../../../utilities/Utilities';
+import { BasePackage } from '../base/BasePackage';
+import { RushConstants } from '../../../RushConstants';
+import { IRushLinkJson } from '../../../data/RushConfiguration';
+import RushConfigurationProject from '../../../data/RushConfigurationProject';
+
+export class PnpmLinkManager extends BaseLinkManager {
+  protected _linkProjects(): Promise<void> {
+    return new Promise<void>((resolve: () => void, reject: (reason: Error) => void): void => {
+      // go ahead and find out the registry we are using, we need this to locate the pnpm store folder
+
+      // EXAMPLE: https://onedrive.pkgs.visualstudio.com/_packaging/odsp-npm/npm/registry
+      const registryUrl: string = Utilities.executeCommandAndCaptureOutput(
```

Comment:
>registryUrl [](start = 12, length = 11)

It's interesting that the PNPM API doesn't provide access to these path names.  Maybe it considers them to be internals?  Maybe we could propose a PR?